relay-metrics/src/aggregation.rs

```diff
@@ -1578,19 +1578,17 @@ impl AggregatorService {
     fn normalize_metric_name(key: &mut BucketKey) -> Result<(), AggregateMetricsError> {
         key.metric_name = match MetricResourceIdentifier::parse(&key.metric_name) {
             Ok(mri) => {
-                if matches!(mri.namespace, MetricNamespace::Unsupported) {
```

Comment:
potential issue: when we had the `unsupported` variant and we failed to parse the namespace we propagated this error in specific, and for other error-cases we propagated `invalidcharacters` error. 

should we let `MetricResoureIdentifier::parse` return `aggregatemetricserror` directly instead? or perhaps create a custom error variant?